docs/guides/src/main/server/configuration.adoc

```diff
@@ -4,202 +4,237 @@
 
-.Set any configuration option to `conf/keycloak.conf`
-```
-db-url-host=keycloak-postgres
-db-username=keycloak
-db-password=change_me
-hostname=mykeycloak.acme.com
-```
+.Defaults
+The production mode sets the following defaults:
 
-.Start the server
-<@kc.start parameters="--optimise"/>
+* HTTP is disabled as transport layer security (HTTPS) is essential
+* Hostname configuration is expected
+* HTTPS/TLS configuration is expected
 
-By using the `keycloak.conf` file, the server can omit some steps at startup. As a result, the server starts faster.
+Make sure to follow the steps outlined in the <@links.server id="configuration-production"/> guide before deploying Keycloak to production environments.
 
-== Configuring the server by using configuration files
+By default, example configuration options for the production mode are commented out in the default `conf/keycloak.conf` file. These give you an idea about the main configuration to consider when running Keycloak in production.
 
-By default, the server always fetches configuration options from the `conf/keycloak.conf` file. For a new installation,
-this file holds only the recommended settings for running in production and those settings are commented out.
+== Setup of the initial admin user
+The initial admin user can be added manually using the web frontend. It needs to be accessed using a local connection (localhost) or using environment variables:
 
-You can also specify a different configuration file by using the `[-cf|--config-file] option by entering the following command:
+To add the initial admin user using environment variables, set `KEYCLOAK_ADMIN=<username>` for the initial admin username and `KEYCLOAK_ADMIN_PASSWORD=<password>` for the initial admin password.
+Keycloak parses these values at first startup to create an initial user with administrative rights.
+Once the first user with administrative rights exists, you can use the admin UI or the command line tool `kcadm.[sh|bat]` to create additional users.
 
-.Running the `build` command using a custom configuration file
-<@kc.build rootParameters="-cf myconfig.conf"/>
+If the initial administrator already exists and the environment variables are still present at startup, an error message stating the failed creation of the initial administrator is shown in the logs. Keycloak ignores the values and starts up correctly.
 
-.Running the `start` command using a custom configuration file
-<@kc.start rootParameters="-cf myconfig.conf"/>
+== Optimize the Keycloak startup
+It is highly recommended to optimize Keycloak for better startup times and memory consumption before deploying into production environments. This section shows you how to apply a set of optimizations for Keycloak to get the best performance and runtime behaviour possible.
```

Comment:
Not sure how important it is to use American spelling, but if its important --> "behavior".

Reply:
changed to behavior in all places, thx. also not entirely sure, but AE is default. I am no native speaker 😅